src/components/Reports/ReportsPageRanking/ReportsPageRanking.js

```diff
@@ -11,31 +11,53 @@ const numberFormatOptions = {
   maximumFractionDigits: 2,
 };
 
+const pageSize = 2;
+const initialState = { pages: [], page: 0, loading: true };
+
 const ReportsPageRanking = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
-  const [state, setState] = useState({ loading: true });
+  const [state, dispatchListEvent] = useReducer((prevState, action) => {
+    switch (action.type) {
+      case 'loadingMore':
+        return { ...prevState, loading: true };
+      case 'moreLoaded':
+        return { pages: [...prevState.pages, ...action.pages], page: prevState.page + 1 };
+      case 'errorOnLoad':
+        return { ...prevState, error: true };
+      default:
+        return initialState;
+    }
+  }, initialState);
+
+  const fetchData = async (datahubClient, domainName, dateFrom, pageNumber) => {
+    const result = await datahubClient.getPagesRankingByPeriod({
+      domainName: domainName,
+      dateFrom: dateFrom,
+      pageSize: pageSize,
+      pageNumber: pageNumber,
+    });
+    if (!result.success) {
+      dispatchListEvent({ type: 'errorOnLoad' });
+    } else {
+      dispatchListEvent({ type: 'moreLoaded', pages: result.value });
+    }
+  };
+
+  const showMoreResults = async () => {
+    dispatchListEvent({ type: 'loadingMore' });
+    fetchData(datahubClient, domainName, dateFrom, state.page + 1);
+  };
```

Comment:
What do you think about moving the dispath line to `fetchData` and simplify this method in the following way: 

```js
  const showMoreResults = () => fetchData(datahubClient, domainName, dateFrom, state.page + 1);
```